improver/regrid/landsea.py

```diff
@@ -256,6 +256,7 @@ def __init__(
         self.output_cube = None
         self.regridder = Nearest(extrapolation_mode=extrapolation_mode)
         self.vicinity = OccurrenceWithinVicinity(vicinity_radius)
+        self._cache_nn = {}
```

Comment:
Could you not do something similar with `functools.cache`?

Reply:
Yes could do if we split the method into two, one with cache returns.
Perhaps neater??  I'll give it a go and see.